pkg/git/utils.go

```diff
@@ -0,0 +1,48 @@
+package git
+
+import (
+	"strings"
+)
+
+// CheckForValidForkRemote checks if the remote URL is a valid remote fork for the upstream URL.
+func CheckForValidForkRemote(upstreamURL, remoteURL, repo string) bool {
+	urlPart1, urlPart2 := extractCommonParts(upstreamURL, remoteURL)
+	if urlPart1 == "https://github.com" && urlPart2 == repo {
+		return true
+	}
+
+	return false
+}
+
+// extractCommonParts takes two Git URLs and returns the common prefix and suffix.
+func extractCommonParts(url1, url2 string) (string, string) {
+	// Split the URLs into segments
+	segments1 := strings.Split(url1, "/")
+	segments2 := strings.Split(url2, "/")
+
+	// Find the common prefix
+	prefixLength := 0
+	for i := 0; i < len(segments1) && i < len(segments2); i++ {
+		if segments1[i] == segments2[i] {
```

Comment:
This and the ones below can be simplified:

```go
if segments1[i] != segments2[i] {
    break
}
prefixLength++
```

Reply:
Fixing